zio-http/shared/src/main/scala/zio/http/endpoint/openapi/JsonSchema.scala

```diff
@@ -254,111 +254,123 @@ object JsonSchema {
       .get
 
   private[openapi] def fromSerializableSchema(schema: SerializableJsonSchema): JsonSchema = {
-    val additionalProperties = schema.additionalProperties match {
-      case Some(BoolOrSchema.BooleanWrapper(bool))  => Left(bool)
-      case Some(BoolOrSchema.SchemaWrapper(schema)) =>
-        val valuesSchema = fromSerializableSchema(schema)
-        Right(
-          schema.optionalKeySchema.fold(valuesSchema)(keySchema =>
-            valuesSchema.annotate(
-              MetaData.KeySchema(
-                fromSerializableSchema(keySchema),
+
+    // if type: object with additionalProperties defined,
+    // but nothing else, we should assume a free form object
+    def anyObject: Boolean = schema.additionalProperties.collect { case BoolOrSchema.BooleanWrapper(true) =>
+      schema.schemaType.contains(TypeOrTypes.Type("object")) &&
+      schema.productIterator.count(_.asInstanceOf[Option[_]].isDefined) == 2
```

Comment:
should probably also allow no type declaration at all, and only `additionalProperties`?

Reply:
this was handled in this commit:
https://github.com/zio/zio-http/pull/3205/commits/d4ef6d4498729ab9e2f209fb014f257a825f1ba0

out of 4 possible cases, 3 work, and one seems to require changes to zio-schema